### PR TITLE
Replace jQuery in "Before You Claim"

### PR DESCRIPTION
**Pull request by mistergone**
This PR replaces jquery functions and methods with "vanilla" JavaScript so we no longer rely on jQuery in this app.

I attempted to replace jQuery without changing any lines of the existing code, but I did have to change the reference to a few event handlers just for ease. The existing functional tests are passing.

## Additions
- Add `dollar-sign.js` which replaces relevant jQuery methods.

## Changes
- Change a few event handlers to use `Query.on()` 

## How to test this PR
1. Check the functional tests in cypress
2. Visually check [the app](http://localhost:8000/consumer-tools/retirement/before-you-claim/) to make sure it matches the [production apa](http://www.consumerfinance.gov/consumer-tools/retirement/before-you-claim/)

## Checklist
- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [x] Project documentation has been updated, potentially one or more of:

### Front-end testing

*This app should function exactly as it did before.*

### CODE REVIEW

File: cfgov/unprocessed/apps/retirement/js/views/graph-view.js [hunk 1]
```diff
@@ -107,7 +106,7 @@ function init() {
     moveIndicatorToAge($(this).attr('data-age-value'));
   });
 
-  $('[data-bar_age]').click(function () {
+  $('#claim-canvas').on('click', '[data-bar_age]', function () {
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/retirement/js/views/graph-view.js:109]: What even is `[data-bar_age]` doing in this call? In the original jquery `on` method docs https://api.jquery.com/on/, it shows the second argument as being passed to the `event.data` object, but this object isn't referenced in the event handler function here, and in fact, the attribute name is looked up independent of the value of this argument. 

Could this whole thing be re-written to something like this?

```js
const claimCanvasDom = document.querySelector('#claim-canvas');
claimCanvasDom.addEventListener('click', function() {
   const age = claimCanvasDom.getAttribute('data-bar_age');
   moveIndicatorToAge(age);
});
```

File: cfgov/unprocessed/apps/retirement/js/views/graph-view.js [hunk 2]
```diff
@@ -192,11 +191,9 @@ function checkEstimateReady() {
...
 function initIndicator() {
-  const $indicator = $('#graph_slider-input');
-
   /* Need both onchange and oninput to work in all browsers
       https://www.impressivewebs.com/onchange-vs-oninput-for-range-sliders/ */
-  $indicator.on('change input', function () {
+  $('#claim-canvas').on('change input', '#graph_slider-input', function () {
     const indicatorValue = Number($(this).val());
     setAgeWithIndicator(indicatorValue);
   });
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/retirement/js/views/graph-view.js:199]: Similar to above, could this be re-written to something like:

```js
const claimCanvasDom = document.querySelector('#claim-canvas');
claimCanvasDom.addEventListener('change', handleAgeIndicator);
claimCanvasDom.addEventListener('input', handleAgeIndicator);

function handleAgeIndicator() {
   const indicatorValue = claimCanvasDom.value();
   setAgeWithIndicator(indicatorValue);
}
```

**Comment by anselmbradford** [comment]
Whoops, gave some suggestions and then realized I mis-read the jquery code for https://api.jquery.com/on/ and had the `selector` and `data` arguments swapped in my head. Deleted suggestions.

**Comment by mistergone** [comment]
It looks like whatever wasn't working in cypress before is working now(?). @wpears - Do you mind looking over this for me?

**Comment by wpears** [comment]
I'm a bit queasy about having two jquery replacements, this one and the one in the `disclosures` app. Is it possible to move this version of dollar-sign to `cfgov/unprocessed/js/modules/util` and use it in both retirement and disclosures?

**Comment by mistergone** [comment]
@wpears - I had some problems with my local environment, but I figured them out, successfully moved the `dollar-sign.js` file to the `util` folder, and now the tests all pass. Hooray!

**Review by wpears — CHANGES_REQUESTED** [review]
~~Getting the following in before you claim:~~
Screenshot 2024-10-16 at 9 06 57 AM

~~and the age adjustment slider isn't operating.~~

**UPDATE** Fixed in [8a3bee2](https://github.com/cfpb/consumerfinance.gov/pull/8321/commits/8a3bee22ebd0ec919307d889840763c46074028e)

**Comment by mistergone** [comment]
> ~Getting the following in before you claim:~ Screenshot 2024-10-16 at 9 06 57 AM
> 
> **UPDATE** Fixed in [8a3bee2](https://github.com/cfpb/consumerfinance.gov/pull/8321/commits/8a3bee22ebd0ec919307d889840763c46074028e)
> 
> But the age adjustment slider isn't operating

I found an incorrect selector that was breaking the slider, and it should be okay now. It was probably a bug from when I was originally writing this PR. We don't have any current functional tests for the slider itself. 😦

**Review by wpears — APPROVED** [review]
:cool:

**Pull request merged**